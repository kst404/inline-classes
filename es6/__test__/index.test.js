import { css, parseStyles } from '../index'

describe('inline-classes', () => {
  describe('parseStyles function', () => {
    it('generate class for simple list of properties', () => {
      const testData = `
        background-color: #fff;
        margin: 0 auto;
      `

      const actual = parseStyles(testData, '._root-selector')

      expect(actual).toMatchSnapshot()
    })

    it('generate rules for selectors', () => {
      const testData = `
        p {
          background-color: #fff;
          margin: 0 auto;
        }
      `

      const actual = parseStyles(testData, '._root-selector')

      expect(actual).toMatchSnapshot()
    })

    it('generate rules for mix of simple list of properties and selectors', () => {
      const testData = `
        width: 50%;
        height: 30vh;

        p {
          background-color: #fff;
          margin: 0 auto;
        }
      `

      const actual = parseStyles(testData, '._root-selector')

      expect(actual).toMatchSnapshot()
    })
  })
})
