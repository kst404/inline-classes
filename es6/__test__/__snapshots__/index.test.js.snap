// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline-classes parseStyles function generate class for simple list of properties 1`] = `
Object {
  "._root-selector": "background-color: #fff; margin: 0 auto;",
}
`;

exports[`inline-classes parseStyles function generate rules for mix of simple list of properties and selectors 1`] = `
Object {
  "._root-selector": "width: 50%; height: 30vh;",
  "p": "background-color: #fff; margin: 0 auto;",
}
`;

exports[`inline-classes parseStyles function generate rules for selectors 1`] = `
Object {
  "p": "background-color: #fff; margin: 0 auto;",
}
`;
